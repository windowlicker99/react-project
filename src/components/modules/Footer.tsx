import React, { FC } from 'react';
import { useSelector } from 'react-redux';
import { Pagination } from '@components/common/Pagination';
import { currentTab, elementAmount, showedAmount } from '@/store/selectors/pagination';

export const Footer: FC = () => {
  const tab = useSelector(currentTab);
  const totalElements = useSelector(elementAmount);
  const elements = useSelector(showedAmount);

  return (
    <footer className="footer">
      <div className="footer-inner ">
        <span className="footer-text">
          Showing {elements} of {totalElements} {tab}
        </span>
        <Pagination />
      </div>
    </footer>
  );
};
