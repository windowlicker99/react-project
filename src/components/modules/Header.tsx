import React, { FC } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from 'react-router';
import { Search } from '@components/common/form-controls/Search';
import { HeaderDropdown } from '@components/common/HeaderDropdown';
import { headerSubtitles, showHeaderArrow } from '@/store/selectors/header';
import { setHeader } from '@/store/slices/headerSlice';
import { CLIENT_PATHS } from '@/constants/paths';

interface IHeaderProps {
  title: string;
}

export const Header: FC<IHeaderProps> = ({ title }) => {
  const dispatch = useDispatch();
  const history = useHistory();
  const showArrow = useSelector(showHeaderArrow);
  const subtitles = useSelector(headerSubtitles);

  const backBtnClick = () => {
    switch (title) {
      case 'Architecture':
        history.push(`${CLIENT_PATHS.sidebar}${CLIENT_PATHS.vehicle}`);
        break;
      default:
        history.goBack();
        break;
    }
    dispatch(setHeader([]));
  };

  return (
    <header className="header">
      <div className="container">
        <nav className="header-nav">
          <div className="header-inner">
            <div className="header-toggler">
              {showArrow ? (
                <span className="icon-arrow-left" onClick={backBtnClick} aria-hidden="true" />
              ) : (
                <span className="icon-menu" />
              )}
            </div>

            <div className="header-title">
              {title}

              {showArrow &&
                subtitles.map((subtitle) => (
                  <span className="header-subtitle" key={subtitle}>
                    {subtitle}
                  </span>
                ))}
            </div>
          </div>
          <div className="header-controls">
            <div className="mr-15">
              <Search formPlaceholder="Search" />
            </div>
            <HeaderDropdown />
          </div>
        </nav>
      </div>
    </header>
  );
};
