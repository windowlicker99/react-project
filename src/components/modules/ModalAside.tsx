import React, { FC, ReactNode } from 'react';
import classNames from 'classnames';

interface IModalAsideProps {
  isOpen: boolean;
  header?: ReactNode;
  footer?: ReactNode;
}

export const ModalAside: FC<IModalAsideProps> = ({ isOpen, header, footer, children }) => (
  <div className={classNames('modal modal-aside', { show: isOpen })}>
    {/* <!-- add class show --> */}
    <div className="modal-content">
      {header && <div className="modal-header">{header}</div>}
      <div className="modal-body">{children}</div>
      {footer && <div className="modal-footer">{footer}</div>}
    </div>
  </div>
);
