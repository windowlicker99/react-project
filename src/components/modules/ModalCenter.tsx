import React, { FC } from 'react';
import classNames from 'classnames';
import { Btn } from '@components/common/form-controls/Btn';

interface IModalCenter {
  title: string;
  subtitle?: string;
  isOpen: boolean;
  disabledApplyBtn?: boolean;
  onClose: () => void;
  onApply: () => void;
}

export const ModalCenter: FC<IModalCenter> = ({
  title,
  subtitle = '',
  disabledApplyBtn = false,
  isOpen,
  onClose,
  onApply,
  children,
}) => (
  <div className={classNames('modal', { show: isOpen })}>
    <div className="modal-content">
      <div className="modal-header">
        <div className="modal-heading">
          <div className="modal-header-content">
            <span className="modal-title">{title}</span>
            <span className="modal-subtitle">{subtitle}</span>
          </div>
          {title !== 'Questionnaire' && (
            <button type="button" className="btn btn-xs-secondary" onClick={onClose}>
              <span className="icon-close" />
            </button>
          )}
        </div>
      </div>
      <div className="modal-body">{children}</div>
      <div className="modal-footer">
        <div className="btn-group-stretch">
          <Btn btnStatus="btn-lg-outline mr-15" btnIcon="" btnText="Cancel" onClick={onClose} />
          <Btn btnStatus="btn-lg-primary" disabledBtn={disabledApplyBtn} btnIcon="" btnText="Apply" onClick={onApply} />
        </div>
      </div>
    </div>
  </div>
);
