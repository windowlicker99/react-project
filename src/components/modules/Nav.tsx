import React, { FC } from 'react';

export const Nav: FC = () => (
  <nav className="nav">
    <ul className="nav-list">
      <li className="nav-item">
        <a href="/" className="nav-link">
          Item 1
        </a>
      </li>
      <li className="nav-item">
        <a href="/" className="nav-link">
          Item 2
        </a>
      </li>
      <li className="nav-item">
        <a href="/" className="nav-link">
          Item 3
        </a>
      </li>
    </ul>
  </nav>
);
