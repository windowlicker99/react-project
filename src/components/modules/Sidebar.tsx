import React, { FC } from 'react';
import { Switch, NavLink, useRouteMatch } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { Logo } from '@components/common/Logo';
import { sidebarTabsList } from '@constants/tabsLists';
import { PageContentLayout } from '@components/page-contents/PageContentLayout';
import { SecurityPage } from '@components/pages/SecurityPage';
import { PrivateRoute } from '@routing/PrivateRoute';
import { setHeader } from '@store/slices/headerSlice';
import { CLIENT_PATHS } from '@constants/paths';

export const Sidebar: FC = () => {
  const { path, url } = useRouteMatch();
  const dispatch = useDispatch();

  return (
    <div className="page-container">
      <nav className="sidebar is-open">
        <div className="sidebar-header">
          <Logo />
        </div>
        <div className="sidebar-body">
          <ul className="sidebar-list">
            {sidebarTabsList.map((tab) => (
              <li className="sidebar-list-item" key={tab.href}>
                <NavLink
                  onClick={() => dispatch(setHeader([]))}
                  to={`${url}${tab.href}`}
                  role={tab.role || ''}
                  data-bs-toggle={tab.dataBsToggle || ''}
                  className="sidebar-link"
                  href={tab.href}
                  data-tab={tab.dataTab || ''}
                >
                  <div className="sidebar-icon">
                    <span className={tab.iconClass} />
                  </div>
                  <span className="sidebar-link-title">{tab.title}</span>
                </NavLink>
              </li>
            ))}
          </ul>
        </div>
      </nav>

      <main className="main">
        <Switch>
          {sidebarTabsList.map((tab) => (
            <PrivateRoute
              exact
              path={`${path}${tab.href}`}
              component={() => <PageContentLayout title={tab.title}>{tab.component({})}</PageContentLayout>}
              key={tab.href}
            />
          ))}
          <PrivateRoute
            exact
            path={CLIENT_PATHS.security}
            component={() => (
              <PageContentLayout title="Security Development">
                <SecurityPage />
              </PageContentLayout>
            )}
          />
        </Switch>
      </main>
    </div>
  );
};
