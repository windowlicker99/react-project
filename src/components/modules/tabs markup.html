<div className="sidebar is-open" id="sidebar">
      <div className="sidebar-header">
        <Logo />
      </div>
      <div className="sidebar-body">
        <ul className="sidebar-list">
          <li className="sidebar-list-item">
            <a
              className="sidebar-link active"
              href="#vehicle"
              data-tab="vehicle"
            >
              <div className="sidebar-icon">
                <span className="icon-vehicle" />
              </div>
              <span className="sidebar-link-title">Vehicle Types</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="#architecture"
              data-tab="architecture"
            >
              <div className="sidebar-icon">
                <span className="icon-architecture" />
              </div>
              <span className="sidebar-link-title">Architecture</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="#elements"
              data-tab="elements"
            >
              <div className="sidebar-icon">
                <span className="icon-elements" />
              </div>
              <span className="sidebar-link-title">Elements</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="#tickets"
              data-tab="tickets"
            >
              <div className="sidebar-icon">
                <span className="icon-tickets" />
              </div>
              <span className="sidebar-link-title">Tickets</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="/#"
            >
              <div className="sidebar-icon">
                <span className="icon-services" />
              </div>
              <span className="sidebar-link-title">Services</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="/#"
            >
              <div className="sidebar-icon">
                <span className="icon-threats" />
              </div>
              <span className="sidebar-link-title">Threats</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="/#"
            >
              <div className="sidebar-icon">
                <span className="icon-controls" />
              </div>
              <span className="sidebar-link-title">Controls</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="/#"
            >
              <div className="sidebar-icon">
                <span className="icon-documents" />
              </div>
              <span className="sidebar-link-title">Documents</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link disabled"
              data-bs-toggle="collapse"
              role="button"
              href="/#"
            >
              <div className="sidebar-icon">
                <span className="icon-reports" />
              </div>
              <span className="sidebar-link-title">Reports</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="#notifications"
              data-tab="notifications"
            >
              <div className="sidebar-icon">
                <span className="icon-notifications" />
              </div>
              <span className="sidebar-link-title">Notifications</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="/#"
            >
              <div className="sidebar-icon">
                <span className="icon-certification" />
              </div>
              <span className="sidebar-link-title">Certification</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="/#"
            >
              <div className="sidebar-icon">
                <span className="icon-feeds" />
              </div>
              <span className="sidebar-link-title">External Feeds</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="/#"
            >
              <div className="sidebar-icon">
                <span className="icon-сonfigurations" />
              </div>
              <span className="sidebar-link-title">Configurations</span>
            </a>
          </li>
          <li className="sidebar-list-item">
            <a
              className="sidebar-link"
              data-bs-toggle="collapse"
              role="button"
              href="/#"
            >
              <div className="sidebar-icon">
                <span className="icon-settings" />
              </div>
              <span className="sidebar-link-title">Settings</span>
            </a>
          </li>
        </ul>
      </div>
    </div>