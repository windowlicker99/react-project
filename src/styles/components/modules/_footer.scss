.footer {
  flex: 0 0 auto;
}

.footer-inner {
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.footer-text {
  font-size: 14px;
  color: $grey;
  font-weight: 500;
}
