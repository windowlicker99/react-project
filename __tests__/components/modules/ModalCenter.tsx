import React from 'react';
import { render } from '@testing-library/react';
import { ModalCenter } from '@components/modules/ModalCenter';

describe('ModalCenter tests', () => {
  it('Should render', () => {
    const { container } = render(<ModalCenter title="Test" isOpen onApply={jest.fn()} onClose={jest.fn()} />);
    expect(container.getElementsByClassName('modal').length).toEqual(1);
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(<ModalCenter title="Test" isOpen onApply={jest.fn()} onClose={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
