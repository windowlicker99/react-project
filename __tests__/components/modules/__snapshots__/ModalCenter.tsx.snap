// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCenter tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal show"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <div
          class="modal-heading"
        >
          <div
            class="modal-header-content"
          >
            <span
              class="modal-title"
            >
              Test
            </span>
            <span
              class="modal-subtitle"
            />
          </div>
          <button
            class="btn btn-xs-secondary"
            type="button"
          >
            <span
              class="icon-close"
            />
          </button>
        </div>
      </div>
      <div
        class="modal-body"
      />
      <div
        class="modal-footer"
      >
        <div
          class="btn-group-stretch"
        >
          <button
            class="btn btn-lg-outline mr-15"
            type="button"
          >
            <span
              class="btn-icon "
            />
            <span
              class="btn-text"
            >
              Cancel
            </span>
          </button>
          <button
            class="btn btn-lg-primary"
            type="button"
          >
            <span
              class="btn-icon "
            />
            <span
              class="btn-text"
            >
              Apply
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
